### PR TITLE
Dispose AnimationController in examples

### LINKED ISSUE #5206: Update animation docs to explain when you should dispose animations
Looking at the flutter source code this seems to be done in [some places](https://github.com/flutter/flutter/blob/3cbb20d176c950effb7d1479661fbe87c8e8ef91/packages/flutter/lib/src/material/toggleable.dart#L201) but [not others](https://github.com/flutter/flutter/blob/b006ca556d7e44c7aafda3a6b5a4637e1000def6/examples/layers/widgets/spinning_square.dart). The [animations page](https://flutter.io/animations/) makes no mention of dispose.

### PR DESCRIPTION
**Pull request by abarth**
These examples should show the best practices for working with
AnimationControllers.

Fixes #5206

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

**Comment by abarth** [comment]
@HansMuller

**Comment by sethladd** [comment]
Oh, great!

**Comment by Hixie** [comment]
LGTM but should we update the docs at the same time?

File: examples/layers/widgets/spinning_square.dart [hunk 1]
```diff
@@ -24,6 +24,12 @@ class _SpinningSquareState extends State<SpinningSquare> {
   }
 
   @override
+  void dispose() {
+    _animation.stop();
```
Comment (HansMuller) [inline examples/layers/widgets/spinning_square.dart]: I thought we agreed (as a convention) to use _animation.dispose() in this situation?

**Comment by HansMuller** [comment]
LGTM tambien

File: examples/layers/widgets/spinning_square.dart (hunk 1 above)
Comment (abarth) [inline examples/layers/widgets/spinning_square.dart]: Done.

**Comment by abarth** [comment]
> should we update the docs at the same time?

The docs are in the web site repo.  I'll either update them or file another bug to track that part of the work.

**Pull request merged**